File: mahimofu.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// 最大公約数を計算する (互助法を使用)
int gcd(int a, int b) {
    while (b != 0) {
        int tmp = b;
        b = a % b;
        a = tmp;
    }
    return a;
}

int main() {
    int max_ans = 100;
    int ans, input_num;

    srand(time(NULL));
    ans = rand() % max_ans + 1;

    while (1) {
        scanf("%d", &input_num);

        if (input_num < 1 || max_ans < input_num){
            printf("1から%dの値を入力してください！\n", max_ans);
            continue;
        }

        if (input_num == ans) {
            printf("まひろもふりすぎてまひもふ!!!!!\n");
            break;
        } else {
            int repeat = gcd(input_num, ans);
            for (int i = 0; i < repeat; i++) {
                printf("まひ");
            }
            printf("\n");
        }
    }
    return 0;
}

File: mahimofu.r
# 互助法を使って最大公約数を求める
gcd <- function(a, b) {
  while(b != 0) {
    temp <- b
    b <- a %% b
    a <- temp
  }
  return(a)
}

# まひもふゲームを実行
mahimofu <- function() {
  max_ans <- 100
  ans <- sample(1:max_ans, 1)

  repeat {
    input_num <- as.integer(readline(prompt = ""))

    if (is.na(input_num) || input_num < 1 || input_num > max_ans) {
      cat("1から", max_ans, "までの値を入力してください！\n")
      next
    }

    if (input_num == ans) {
      cat("まひろもふりすぎてまひもふ!!!!!\n")
      break
    } else {
      cat(paste0(rep("まひ", gcd(input_num, ans)), collapse = ""), "\n")
    }
  }
}

# これより上の部分をコピペし、
# mahimofu() と入力してゲーム開始

if (interactive()) {
  mahimofu()
}